pyfstat/core.py

```diff
@@ -982,23 +1065,82 @@ def get_fullycoherent_twoF(
             self.PulsarDopplerParams.argp = float(argp)
 
         lalpulsar.ComputeFstat(
-            self.FstatResults,
-            self.FstatInput,
-            self.PulsarDopplerParams,
-            1,
-            self.whatToCompute,
+            Fstats=self.FstatResults,
+            input=self.FstatInput,
+            doppler=self.PulsarDopplerParams,
+            numFreqBins=1,
+            whatToCompute=self.whatToCompute,
         )
+        # We operate on a single frequency bin, so we grab the 0 component
+        # of what is internally a twoF array.
+        self.twoF = np.float(self.FstatResults.twoF[0])
+        return self.twoF
 
-        if not self.transientWindowType:
-            if self.BSGL is False:
-                return self.FstatResults.twoF[0]
+    def get_fullycoherent_single_IFO_twoFs(self):
+        """Computes single-detector F-stats at a single point.
 
-            twoF = np.float(self.FstatResults.twoF[0])
-            self.twoFX[0] = self.FstatResults.twoFPerDet(0)
-            self.twoFX[1] = self.FstatResults.twoFPerDet(1)
-            log10BSGL = lalpulsar.ComputeBSGL(twoF, self.twoFX, self.BSGLSetup)
-            return log10BSGL
+        This requires `self.get_fullycoherent_twoF()` to be run first.
 
+        Returns
+        -------
+        twoFX: list
+            A list of the single-detector detection statistics twoF.
+            Also stored as `self.twoFX`.
+        """
+        self.twoFX[: self.FstatResults.numDetectors] = [
+            self.FstatResults.twoFPerDet(X)
+            for X in range(self.FstatResults.numDetectors)
+        ]
+        return self.twoFX
```

Comment:
Suggestions:

1) Make it private
2) raise meaningful exception if `self.get_fullycoherent_twoF` was not run.
3) Make it run `self.get_fullycoherent_twoF` (problems?)

Reply:
>    Make it private

I've thought about it, but there may be use cases where one wants to compute these post-hoc. On the other hand, it will always just work on the last grid point, so such cases would be fairly limited, and maybe we should just decide to discourage users from single-point operations on search classes in general?

>   raise meaningful exception if self.get_fullycoherent_twoF was not run.

Good idea, but complicated by the fact that "not run" could mean both "never run" and "only run on previous grid points, not the current one". Which would be an argument in favour of "discourage users from single-point operations on search classes", because then one could safely assume that the latter doesn't happen.

>    Make it run self.get_fullycoherent_twoF (problems?)

Then the structure of `get_fullycoherent_detstat` would need to be changed to not double-call it, and would be less intuitive.

Reply:
@Rodrigo-Tenorio any further thoughts on this aspect?

Reply:
Read-the-docs philosophy: Leave as is

Reply:
issue for future consideration -> #262